app/lib/features/space/pages/chats_page.dart

```diff
@@ -25,153 +26,184 @@ class SpaceChatsPage extends ConsumerWidget {
                     ),
                   ),
-                );
-              }
-              return SliverToBoxAdapter(
-                child: Center(
-                  heightFactor: 1,
-                  child: EmptyState(
-                    title: 'No chats in this space yet',
-                    subtitle:
-                        'Get the conversation going to start organizing collaborating',
-                    image: 'assets/images/empty_chat.svg',
-                    primaryButton: canCreateSpace
-                        ? ElevatedButton(
-                            onPressed: () => context.pushNamed(
-                              Routes.createChat.name,
-                              queryParameters: {'spaceId': spaceIdOrAlias},
-                              extra: 1,
-                            ),
-                            child: const Text('Create Space Chat'),
-                          )
-                        : null,
+                  PopupMenuItem(
+                    onTap: () => context.pushNamed(
+                      Routes.linkChat.name,
+                      pathParameters: {'spaceId': spaceIdOrAlias},
+                    ),
+                    child: const Row(
+                      children: <Widget>[
+                        Text('Link existing Chat'),
+                        Spacer(),
+                        Icon(Atlas.chats),
+                      ],
+                    ),
                   ),
-                ),
-              );
-            },
-            error: (error, stackTrace) => SliverToBoxAdapter(
-              child: Center(child: Text('Failed to load events due to $error')),
-            ),
-            loading: () => const SliverToBoxAdapter(
-              child: Center(child: CircularProgressIndicator()),
-            ),
+                ],
+              ),
+            ],
           ),
-          related.when(
-            // FIXME: filter these for entries we are already members of
-            data: (spaces) =>
-                RiverPagedBuilder<Next?, SpaceHierarchyRoomInfo>.autoDispose(
-              firstPageKey: const Next(isStart: true),
-              provider: remoteChatHierarchyProvider(spaces),
-              itemBuilder: (context, item, index) =>
-                  ConvoHierarchyCard(space: item),
-              noItemsFoundIndicatorBuilder: (context, controller) =>
-                  const SizedBox.shrink(),
-              pagedBuilder: (controller, builder) => PagedSliverList(
-                pagingController: controller,
-                builderDelegate: builder,
+        );
+      },
+      orElse: () => const SliverToBoxAdapter(child: SizedBox.shrink()),
+    );
+  }
+
+  Widget _renderEmpty(BuildContext context, WidgetRef ref) {
+    final chats = ref.watch(relatedChatsProvider(spaceIdOrAlias));
+
+    if ((chats.valueOrNull?.isNotEmpty ?? true)) {
+      // we are still loading or chats has been found locally
+      return Container();
```

Comment:
Why not use ```const SizedBox()```?

Reply:
I saw @kumarpalsinh25 use this one and honestly don't know if there is any reason to prefer one of the other... Is there?

Reply:
> I saw @kumarpalsinh25 use this one and honestly don't know if there is any reason to prefer one of the other... Is there?

Pretty sure, that might be overlooked. Flutter  recommends using ```const SizedBox()``` for empty UI instead of ```Container()```.

Reply:
@gtalha07 do you have a link/reference for that? I'd like to know more about it ...

Reply:
> @gtalha07 do you have a link/reference for that? I'd like to learn more about it ...

Indeed, here are some sources which talks on the usage of ```Container``` vs SizedBox.
https://medium.com/@olasoji.od/flutter-widgets-sizedbox-vs-container-7d9cf3e566f7
https://stackoverflow.com/questions/57703182/flutter-container-vs-sizedbox-for-dummy-empty

> Both of these widgets will create an empty space with the specified width and height. The SizedBox.shrink constructor is useful when you want to create a zero-sized box, which can be useful for spacing or padding. The Container widget can also be used for empty space, but it provides more customization options such as color, margin, padding, and decoration.

> It's worth noting that SizedBox is generally recommended for creating empty space because it's more lightweight and performant than Container. However, Container can be a better choice if you need more customization options or if you're already using it for other purposes.

Reply:
> I saw @kumarpalsinh25 use this one and honestly don't know if there is any reason to prefer one of the other... Is there?

@gnunicorn 
My bad habit! Not major difference but still I agree with @gtalha07 that SizedBox is good for empty placeholder use-case.